core/src/main/java/hudson/Util.java

```diff
@@ -545,25 +543,17 @@ public static boolean isRelativePath(String path) {
      * @see InvalidPathException
      */
     public static boolean isDescendant(File forParent, File potentialChild) throws IOException {
-        try {
-            Path child = potentialChild.getAbsoluteFile().toPath().normalize();
-            Path parent = forParent.getAbsoluteFile().toPath().normalize();
-            return child.startsWith(parent);
-        } catch (InvalidPathException e) {
-            throw new IOException(e);
-        }
+        Path child = fileToPath(potentialChild.getAbsoluteFile()).normalize();
+        Path parent = fileToPath(forParent.getAbsoluteFile()).normalize();
+        return child.startsWith(parent);
     }
 
     /**
      * Creates a new temporary directory.
      */
     public static File createTempDir() throws IOException {
-        File tmp = File.createTempFile("jenkins", "tmp");
-        if(!tmp.delete())
-            throw new IOException("Failed to delete "+tmp);
-        if(!tmp.mkdirs())
-            throw new IOException("Failed to create a new directory "+tmp);
-        return tmp;
+        // https://stackoverflow.com/questions/617414/how-to-create-a-temporary-directory-folder-in-java
+        return Files.createTempDirectory("jenkins").toFile();
```

Comment:
I believe that on POSIX systems `Files.createTempDirectory` will create a directory with `0700&(~umask)` permissions while the previous code will create a directory with `0777&(~umask)` permissions. Would be good to confirm to prevent another [JENKINS-48407 ](https://issues.jenkins-ci.org/browse/JENKINS-48407).

Reply:
Hello Devin,

I checked and you are correct; Files.createTempDirectory() does use permissions `0700&(~umask)` for the tmp dir.

I found PR #3161 whose approach could be used here. I will add a commit to do this.

Reply:
New commit added. Would you please take a look?